tuning/autotune.py

```diff
@@ -673,6 +762,69 @@ def sort_candidates_by_first_benchmark_times(
     return sorted_indexes
 
 
+def group_benchmark_results_by_device_id(
+    benchmark_results: list[TaskResult],
+) -> list[list[TaskResult]]:
+    """
+    Groups benchmark results by device ID.
+
+    e.g.
+    [TaskResult(res1, device_1), TaskResult(res2, device_2), TaskResult(res3, device_1)]
+    ----->
+    [ [TaskResult(res1, device_1), TaskResult(res3, device_1)], [TaskResult(res2, device_2)] ]
+    """
+    grouped_results = [
+        list(group)
+        for _, group in groupby(benchmark_results, key=lambda tr: tr.device_id)
+    ]
+    grouped_results: dict[int, list[TaskResult]] = {}
+    for result in benchmark_results:
+        if result.device_id not in grouped_results:
+            grouped_results[result.device_id] = []
+        grouped_results[result.device_id].append(result)
+
+    grouped_benchmark_results = [
+        grouped_results[device_id] for device_id in sorted(grouped_results)
+    ]
+
+    return grouped_benchmark_results
+
+
+def parse_grouped_benchmark_results(
+    grouped_benchmark_results: list[list[TaskResult]],
+    candidate_trackers: CandidateTracker,
+) -> list[str]:
+    """Update candidate_trackers and and collect strings"""
+    dump_list = []
+
+    for same_device_results in grouped_benchmark_results:
+        for unet_candidate_result in same_device_results:
+            res = BenchmarkOutput(unet_candidate_result.result.stdout)
+            if "unet_baseline.vmfb" in res.unet_candidate_path:
+                baseline_time = res.benchmark_time
+                dump_str = res.output_str
```

Comment:
```suggestion
                dump_list.append(res.output_str)
                continue
```

and then you won't need the `else` below. See https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html and https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code